### PR TITLE
Add infix operators for mapOk/flatMapOk

### PR DESCRIPTION
**Pull request by johnhaley81**
I wanted to see at least what the interest would be in adding some infix operators to the codebase. I chose what I would think are the 2 most common methods to add them to and I lifted the operators from Haskell.

### CODE REVIEW

**Comment by gilbert** [comment]
I don't have any strong opinions. Any thoughts @scull7 @ostera?

File: tests/TestFuture.re [hunk 1]
```diff
@@ -118,11 +118,10 @@ describe("Future", () => {
 
 
 describe("Future Belt.Result", () => {
+  let ((>>=), (<$>)) = Future.((>>=), (<$>));
 
   test("mapOk", () => {
-    Belt.Result.Ok("two")
-    |. Future.value
-    |. Future.mapOk(s => s ++ "!")
+    (Belt.Result.Ok("two") |. Future.value <$> (s => s ++ "!"))
```
Comment (scull7) [inline tests/TestFuture.re]: If we're going to provide the infix operators they should come with their own test and not replace the standard tests.  Tests are documentation, and having the infix operators in their own place would induce much less confusion.

File: tests/TestFuture.re [hunk 2]
```diff
@@ -154,9 +153,10 @@ describe("Future Belt.Result", () => {
   });
 
   test("flatMapOk", () => {
-    Belt.Result.Ok("four")
-    |. Future.value
-    |. Future.flatMapOk(s => Belt.Result.Ok(s ++ "!") |. Future.value)
+    (
+      Belt.Result.Ok("four") |. Future.value
+      >>= (s => Belt.Result.Ok(s ++ "!") |. Future.value)
```
Comment (scull7) [inline tests/TestFuture.re]: I'm not sure this add clarity to the code here. (Also, same admonishment as above.)  Perhaps there's a more clear example which shows the ergonomic gains of the infix operators.

**Review by scull7 — CHANGES_REQUESTED** [review]

**Comment by johnhaley81** [comment]
This might be out of scope, but running `npm run test:only` returns an error locally and it looks like it's also [doing the same in CI](https://travis-ci.org/RationalJS/future/jobs/459866018#L1515).

This is without my changes.

**Comment by scull7** [comment]
@johnhaley81 I've opened an issue so the CI failure can be addressed separately

File: tests/TestFuture.re [hunk 3]
```diff
@@ -262,4 +263,42 @@ describe("Future Belt.Result", () => {
...
   });
 
+  test("<$>", () => {
+    Future.value(Belt.Result.Ok("infix ops"))
+    <$> ((++)(" "))
+    <$> ((++)("rock!"))
+    |. Future.get(r =>
+      r
+      |. Belt.Result.getExn
+      |. equals("infix ops rock!2")
```
Comment (scull7) [inline tests/TestFuture.re]: Where does the `2` come from?

Comment (johnhaley81) [reply tests/TestFuture.re]: 🤦‍♂️ from me leaving that in to see if the test would fail (it didn't) when I was researching #29.

I'll change that.

**Comment by johnhaley81** [comment]
@scull7 are these changes sufficient or am I missing something?

**Review by scull7 — APPROVED** [review]

**Comment by scull7** [comment]
Sorry it took me so long to get back to this.

**Comment by johnhaley81** [comment]
No worries. Thanks!

**Comment by gilbert** [comment]
Thanks all!

**Pull request merged**